operations/configuration_management/CFS_Configurations.md

```diff
@@ -0,0 +1,248 @@
+This is necessary so that CFS can easily monitor the desired state of components.
+
+(`ncn-mw#`) In the following example, when the configuration is created or updated, CFS will automatically check with VCS to get the commit at the head of the branch.
+Both the commit and the branch are then stored. The commit acts as normal, and the branch is stored to make future updates to the commit easier.
+
+```bash
+cat configurations-example.json
+```
+
+Example configuration:
+
+```json
+{
+  "layers": [
+    {
+      "name": "configurations-layer-example-1",
+      "clone_url": "https://api-gw-service-nmn.local/vcs/cray/example-repo.git",
+      "playbook": "site.yml",
+      "branch": "main"
+    }
+  ]
+}
+```
+
+```bash
+cray cfs v3 configurations update configurations-example \
+--file ./configurations-example.json \
+--format json
+```
+
+Example output:
+
+```json
+{
+  "last_updated": "2021-07-28T03:26:30:37Z",
+  "layers": [
+    {
+      "clone_url": "https://api-gw-service-nmn.local/vcs/cray/example-repo.git",
+      "commit": "43ecfa8236bed625b54325ebb70916f55884b3a4",
+      "branch": "main",
+      "name": "configurations-layer-example-1",
+      "playbook": "site.yml"
+    }
+  ],
+  "name": "configurations-example"
+}
+```
+
+(`ncn-mw#`) If changes are made to a repository and branches are specified in the configuration,
+users can then use the `--update-branches` flag to update a configuration so that all commits reflect the latest commit on the branches specified.
+
+```bash
+cray cfs v3 configurations update configurations-example --update-branches
+```
+
+Example output:
+
+```json
+{
+  "last_updated": "2021-07-28T03:26:30:37Z",
+  "layers": [
+    {
+      "clone_url": "https://api-gw-service-nmn.local/vcs/cray/example-repo.git",
+      "commit": "<latest git commit id>",
+      "branch": "main",
+      "name": "configurations-layer-example-1",
+      "playbook": "site.yml"
+    }
+  ],
+  "name": "configurations-example"
+}
+```
+
+## Create a CFS Configuration
+
+Use the `cray cfs v3 configurations update` command to create a configuration.
+
+1. (`ncn-mw#`) Create a JSON file to hold data about the CFS configuration.
+
+   ```bash
+   cat example-config.json
+   ```
+
+   Example configuration:
+
+   ```json
+   {
+    "description": "this is an optional field",
+    "layers": [
+      {
+        "name": "configurations-layer-example-1",
+        "clone_url": "https://api-gw-service-nmn.local/vcs/cray/example-repo.git",
+        "playbook": "site.yml",
+        "commit": "<git commit id>"
+      },
+      {
+        "name": "configurations-layer-example-2",
+        "clone_url": "https://api-gw-service-nmn.local/vcs/cray/example-repo2.git",
+        "playbook": "site.yml",
+        "commit": "<git commit id>"
+      },
+      # { ... add more configuration layers here, if needed ... }
+    ]
+   }
+   ```
```

Comment:
This was specifically requested because it wasn't clear to a customer I believe.  Any idea how to leave the comment there and avoid a formatting error?

Reply:
Well, that's why I added the comment after the example, instead of inside of it. 

There is no way to add a proper comment string in JSON. If adding a comment outside of the example block (like I suggested) isn't desirable, then the options are:
* Specify the block as text format (which will avoid the error, but will also make it look less nice in some browsers)
* Make the example use valid JSON that includes the comment text, but at the cost of making it no longer a legal CFS example:
```json
{
  "description": "example playbook",
  "layers": [
    {
      "name": "configurations-layer-example-1",
      "clone_url": "https://api-gw-service-nmn.local/vcs/cray/example-repo.git",
      "playbook": "site.yml",
      "commit": "43ecfa8236bed625b54325ebb70916f55884b3a4"
    },
    "Add more configuration layers here, if needed"
  ]
}
```

I think my suggestion is the best of these three options, but I'm flexible.